src/main/scala/com/qubole/spark/hiveacid/reader/v2/HiveAcidColumnVector.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.qubole.spark.hiveacid.reader.v2;
+
+import java.math.BigDecimal;
+
+import com.qubole.shaded.hadoop.hive.ql.exec.vector.*;
+
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.types.TimestampType;
+import org.apache.spark.sql.types.LongType;
+import org.apache.spark.sql.types.DoubleType;
+import org.apache.spark.sql.types.ByteType;
+import org.apache.spark.sql.types.DecimalType;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+import com.qubole.shaded.hadoop.hive.ql.exec.vector.ColumnVector;
+
+/**
+ * A column vector class wrapping Hive's ColumnVector. Because Spark ColumnarBatch only accepts
+ * Spark's vectorized.ColumnVector, this column vector is used to adapt Hive ColumnVector with
+ * Spark ColumnarVector. This class is a copy of spark ColumnVector which is declared private.
+ */
+public class HiveAcidColumnVector extends org.apache.spark.sql.vectorized.ColumnVector {
```

Comment:
This looks similar to org.apache.spark.sql.execution.datasources.orc.OrcColumnVector. Should we not be extending it or using it directly here. We should also be Format agnostic but we can use something like Factory etc to get ColumnVector based on format.

Reply:
OrcColumnVector in spark is in private scope so had to copy it.

Reply:
this file is removed